tests/integ/scala/test_dataframe_writer_suite.py

```diff
@@ -139,6 +135,56 @@ def test_negative_write_with_target_column_name_order(session):
         session.table(table_name).drop_table()
 
 
+def test_write_with_target_column_name_order_all_kinds_of_dataframes_without_truncates(
+    session,
+):
+    table_name = Utils.random_table_name()
+
+    def create_table():
+        session.create_dataframe(
+            [],
+            schema=StructType(
+                [StructField("a", IntegerType()), StructField("b", IntegerType())]
+            ),
+        ).write.save_as_table(table_name, table_type="temporary")
+
+    create_table()
+    try:
+        df1 = session.create_dataframe([[1, 2]], schema=["b", "a"])
+        # DataFrame.cache_result()
+        df_cached = df1.cache_result()
```

Comment:
Question: why do we need cache_result here?

Reply:
We actually don't for the scope of this change. This was part of the skipped test case that I reused. Can remove it.